### DISCUSSION TITLE
Running GUI apps on client which has no physical screen wont work

### DISCUSSION THREAD

**Discussion by DLangenberg** [devcontainers/community#180] [General]
At our company the developers develop on a remote windows 11 machine which has only a lan and power cable attached. When we use the WSL2 engine in docker in combination with DevContainer in vscode, gui apps cant run as they do not recognise a screeb but if i start a different WSL image i can run gui apps.

Setting the $DISPLAY variable to :0 doesnt make a difference, i think i am missing something in the setup but i am sure we must not be the only one having this issue?

i would normally share my devcontainer.json but even with default images i have this issue, so i dont feel this is related to my config

The moment i connect a screen and reboot the machine i can use GUI apps like xclock again, but this is not a solution for us.

**Comment by chrmarti** [comment]
When you connect to the remote machine using the Remote-SSH extension and then connect to a Dev Container it automatically forwards X11 if DISPLAY is set locally. Please check the Dev Containers log after connecting to the dev container. (`F1` > `Dev Containers: Show Container Log`)

Maybe you are connecting to the remote machine with Remote Desktop?
